stripe/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowControllerInitializer.kt

```diff
@@ -129,6 +130,26 @@ internal class DefaultFlowControllerInitializer(
         )
     }
 
+    private suspend fun setLastSavedPaymentMethod(
+        prefsRepository: PrefsRepository,
+        isGooglePayReady: Boolean,
+        paymentMethods: List<PaymentMethod>
+    ) {
+        if (prefsRepository.getSavedSelection() == SavedSelection.None) {
+            paymentMethods.firstOrNull()?.let { paymentMethod ->
+                prefsRepository.savePaymentSelection(
+                    PaymentSelection.Saved(paymentMethod)
+                )
+            }
+        }
+
+        if (isGooglePayReady &&
+            prefsRepository.getSavedSelection() == SavedSelection.None
+        ) {
+            prefsRepository.savePaymentSelection(PaymentSelection.GooglePay)
+        }
```

Comment:
`prefsRepository.savePaymentSelection` seems to be writing a [single](https://github.com/stripe/stripe-android/blob/32fdef14dd28d8d324d45d7d84f75de21a1b7c58/stripe/src/main/java/com/stripe/android/paymentsheet/DefaultPrefsRepository.kt#L51) shared pref, we could possibly avoid a rewrite by collapsing both if checks, also probably inline this function too as it's not used else where

```suggestion
        if (prefsRepository.getSavedSelection() == SavedSelection.None) {
            if(isGooglePayReady) {
                prefsRepository.savePaymentSelection(PaymentSelection.GooglePay)
            } else {
                paymentMethods.firstOrNull()?.let { paymentMethod ->
                    prefsRepository.savePaymentSelection(
                        PaymentSelection.Saved(paymentMethod)
                    )
                }
            }
        }
```

Reply:
Can you clarify what you mean by: we could possibly avoid a rewrite by collapsing both if checks?

If you feel strongly this should be inside the method I can do that.  I personally feel like it improves readability.

Reply:
Oh I meant `prefsRepository.savePaymentSelection` might be called twice if we align two `if` blocks, which ended up overwriting the same sharedpref. We could avoid that by branching off of the block and only write it once. Don't have any preference on inline the entire method though :)

Reply:
Preferences are all cached, so I am not sure it will have a huge performance impact.

Reply:
hmm I believe it's more about readability over performance impact:) With a if-else check, the reader could tell the function is deciding only _1_ single payment selection, while two non-exclusive if checks might imply the impl of `prefsRepository#savePaymentSelection` is aggregative.

Reply:
Agree that the proposed change looks clearer.